src/EnergyPlus/HVACVariableRefrigerantFlow.cc

```diff
@@ -9355,9 +9359,13 @@ void VRFTerminalUnitEquipment::ControlVRFToLoad(EnergyPlusData &state,
         }
     } else if (VRFCoolingMode || HRCoolingMode) {
         // IF the system is in cooling mode and/or the terminal unit requests cooling
-        if (NoCompOutput <= QZnReq) {
-            PartLoadRatio = 0.0;
-            return;
+        if ((NoCompOutput <= QZnReq && QZnReq < -DataHVACGlobals::SmallLoad) ||
+            (QZnReq >= -DataHVACGlobals::SmallLoad && NoCompOutput <= QZnReq && !HRCoolingMode)) {
+            state.dataHVACVarRefFlow->VRFTU(VRFTUNum).coolingCoilActive = false;
+            if (!this->SuppHeatingCoilPresent) {
+                PartLoadRatio = 0.0;
+                return;
+            }
```

Comment:
The first comparison looks correct, there is excess cooling from the fan operation and there is a cooling load. For the or comparitor you want to fall through if there is a valid heating load, which it looks like this does (after changing to a +SmallLoad).
 
    if ((NoCompOutput <= QZnReq && QZnReq < -DataHVACGlobals::SmallLoad) ||
            (QZnReq >= DataHVACGlobals::SmallLoad && NoCompOutput <= QZnReq && !HRCoolingMode)) {

That just leaves whether !HRCoolingMode is needed. This is inside `} else if (VRFCoolingMode || HRCoolingMode) {` so that means 4 possibilities. The first (no HR) means this system is not capable of heat recovery and second (HR) means it is.

    VRFCoolingMode && !HRCoolingMode <-- system is in cooling mode and cannot run the DX heating coil, but supp heat can run, or system is in HRHeatingMode which will be caught by a different IF block.
    VRFCoolingMode && HRCoolingMode <-- system is in cooling mode and can run the DX cooling coil so DO NOT drop into this IF block
    HRCoolingMode && !HRCoolingMode <-- not possible
    HRCoolingMode && HRCoolingMode <-- redundant

So it looks like only the 1st case needs to drop into this IF block. It does look like !HRCoolingMode is needed here (better review my logic). These conditionals are hard to flesh out with all the allowed configurations.

I do see NoCompOutput <= QZnReq on both sides of the or symbol which means this can be simplified.

    if (NoCompOutput <= QZnReq && (QZnReq < -DataHVACGlobals::SmallLoad) ||
            (QZnReq >= DataHVACGlobals::SmallLoad && !HRCoolingMode)) {

Reply:
I agree, added the simplified suggested.

Reply:
I spent some time looking through Init to see how these 4 variables are set.

    VRFCoolingMode = state.dataHVACVarRefFlow->CoolingLoad(VRFCond);
    VRFHeatingMode = state.dataHVACVarRefFlow->HeatingLoad(VRFCond);
    HRCoolingMode = state.dataHVACVarRefFlow->TerminalUnitList(TUListIndex).HRCoolRequest(IndexToTUInTUList);
    HRHeatingMode = state.dataHVACVarRefFlow->TerminalUnitList(TUListIndex).HRHeatRequest(IndexToTUInTUList);

Then I looked at this outer IF block you are modifying.

    if (VRFCoolingMode && HRHeatingMode) {
        // IF the system is in cooling mode, but the terminal unit requests heating (heat recovery)
        if (NoCompOutput >= QZnReq) return;
    } else if (VRFHeatingMode && HRCoolingMode) {
        // IF the system is in heating mode, but the terminal unit requests cooling (heat recovery)
        if (NoCompOutput <= QZnReq) return;
    } else if (VRFCoolingMode || HRCoolingMode) {
        // IF the system is in cooling mode and/or the terminal unit requests cooling
        if (NoCompOutput <= QZnReq) return;
    } else if (VRFHeatingMode || HRHeatingMode) {
        // IF the system is in heating mode and/or the terminal unit requests heating
        if (NoCompOutput >= QZnReq) return;
    }

This block simply says if there is too much capacity without the coils, don't turn on the coils.

The only case I see here that is missing is if the system is in cooling mode (VRFCoolingMode = true) and there is a zone heating load (QZnReq > SmallLoad) and supplemental heater. The HR case is already caught in the first IF. So I see this system cooling mode with zone heating load case as:

    } else if (VRFCoolingMode || HRCoolingMode) {
        // IF the system is in cooling mode and/or the terminal unit requests cooling
        if (QZnReq < -SmallLoad) { <-- the supp heater will not turn on
            if (NoCompOutput <= QZnReq) return;
        // IF the system is in cooling mode and/or the terminal unit requests heating
        } elseif (QZnReq > SmallLoad) { <-- the supp heater might turn on, if one exists
            state.dataHVACVarRefFlow->VRFTU(VRFTUNum).coolingCoilActive = false;
            if (NoCompOutput >= QZnReq) return;
       }

Isn't it as simple as that? Does it even matter if there is a supp heating coil? If not, then nothing should happen when this falls through, however, the logic below was written when there was no supp heating coil allowed. Of course I haven't tested this.

I don't mean to make this harder than it should be, but I do know that getting this logic right is pretty important.